### PR TITLE
Patch gostream release bug for modules

### PR DESCRIPTION
**Pull request by hexbabe**
https://viaminc.slack.com/archives/C03Q69SCFDK/p1724870585927499

### CODE REVIEW

**Review by randhid — APPROVED** [review]

**Review by cheukt — APPROVED** [review]

**Comment by cheukt** [comment]
maybe add a test to prevent regressions?

**Review by cheukt — CHANGES_REQUESTED** [review]
^

**Comment by randhid** [comment]
@hexbabe `media.go` seems to be well tested, but I suspect the issues you saw stem from `stream.go`,  which is virtually untested. Would you have a ballpark on the effort required to do this versus work on the rtsp pooling issues?

**Comment by randhid** [comment]
I'm okay with keeping tests as they are if we don't know what the root of the issue actually is.

**Comment by hexbabe** [comment]
> maybe add a test to prevent regressions?

Thanks for the suggestion! We peeled back one more layer: the reason for this deref panic is because for some reason in the face-registration cam code path, sometimes we spin up a stream and call .Next(), sometimes we just spin up a stream and close it immediately without calling .Next(). When the latter case follows the former case, we deref the `current` from the former case without the expected reffing from a Next() call. Reading the test will make this clearer.

File: gostream/media_test.go [hunk 1]
```diff
@@ -191,3 +191,25 @@ func TestStreamMultipleConsumers(t *testing.T) {
...
+	test.That(t, err, test.ShouldBeNil)
+	// Get one frame
+	_, release, err := stream.Next(context.Background())
+	test.That(t, err, test.ShouldBeNil)
+	release()
+	// Close stream
+	stream.Close(context.Background())
+
+	// Spin up stream and close without calling Next
+	stream, err = videoSrc.Stream(context.Background())
```
Comment (hexbabe) [inline gostream/media_test.go:212]: This will cause a panic without the change in `media.go`

**Comment by hexbabe** [comment]
> We peeled back one more layer

The next layer to peel (if we want to) is why we are spinning up streams, then closing them without calling .Next()

**Comment by hexbabe** [comment]
Screenshot 2024-08-29 at 1 17 59 PM

Tested multiple streams with the face registration module. Streaming the webcam and the face reg stream at the same time.

Are there any other manual tests I should do @cheukt ?

**Review by cheukt — APPROVED** [review]
LGTM! thanks for adding the test, manual tests seem sufficient as well

**Pull request merged**